### PR TITLE
feat(learn): add blockType to block meta

### PR DESCRIPTION
**Pull request by moT01**
This adds the ability to add a blockType to the block meta.json files - so we can label the blocks.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

[ref](https://github.com/freeCodeCamp/fCC10/issues/12)

### CODE REVIEW

File: curriculum/schema/challenge-schema.js [hunk 1]
```diff
@@ -85,6 +85,7 @@ const schema = Joi.object()
   .keys({
     block: Joi.string().regex(slugRE).required(),
     blockId: Joi.objectId(),
+    blockType: Joi.valid('workshop', 'lab', 'lecture', 'quiz', 'exam'),
```
[the remark this answers is not in the document]
Comment (moT01) [reply curriculum/schema/challenge-schema.js:88]: I like it, but the existing blocks don't have a type, so we would need to go add that to all of them. Some of which, may not fit into one of these categories.

[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply curriculum/schema/challenge-schema.js:88]: That's true. I'm happy to take a look and see if there are any that don't seem to fit.

Also, we can do that in another PR if you like.

[the remark this answers is not in the document]
Comment (moT01) [reply curriculum/schema/challenge-schema.js:88]: We could probably do a `when(superblock is front-end..).required`

What do you think about that?

[the remark this answers is not in the document]
Comment (moT01) [reply curriculum/schema/challenge-schema.js]: Well, I went ahead and added that. I think it's the path of least resistance for now. Let me know if you think we should do something else.

[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply curriculum/schema/challenge-schema.js]: Works for me. We can look at the rest after merging.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM 👍

**Review by Sembauke — APPROVED** [review]

**Comment by Sembauke** [comment]
@moT01 Could you resolve the merge conflicts?

**Pull request merged**